dora/tests/src/test/java/alluxio/client/fs/DoraFileSystemIntegrationTest.java

```diff
@@ -0,0 +1,169 @@
+          .setProperty(PropertyKey.USER_FILE_RESERVED_BYTES, Constants.MB * 16 / 2)
+          .setProperty(PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED, true)
+          .setProperty(PropertyKey.WORKER_BLOCK_STORE_TYPE, "PAGE")
+          .setProperty(PropertyKey.WORKER_PAGE_STORE_PAGE_SIZE, Constants.KB)
+          .setProperty(PropertyKey.WORKER_PAGE_STORE_SIZES, "1GB")
+          .setProperty(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.NOOP)
+          .setProperty(PropertyKey.UNDERFS_S3_ENDPOINT, "localhost:8001")
+          .setProperty(PropertyKey.UNDERFS_S3_ENDPOINT_REGION, "us-west-2")
+          .setProperty(PropertyKey.UNDERFS_S3_DISABLE_DNS_BUCKETS, true)
+          .setProperty(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS, "s3://" + TEST_BUCKET)
+          .setProperty(PropertyKey.DORA_CLIENT_UFS_ROOT, "s3://" + TEST_BUCKET)
+          .setProperty(PropertyKey.S3A_ACCESS_KEY, mS3Proxy.getAccessKey())
+          .setProperty(PropertyKey.S3A_SECRET_KEY, mS3Proxy.getSecretKey())
+          .setNumWorkers(1)
+          .setStartCluster(false)
+          .build();
+
+  private static final String TEST_BUCKET = "test-bucket";
+  private static final String TEST_FILE = "test-file";
+  private static final AlluxioURI TEST_FILE_URI = new AlluxioURI("/" + "test-file");
+  private static final String TEST_CONTENT = "test-content";
+  private static final String UPDATED_TEST_CONTENT = "test-content";
+
+  private FileSystem mFileSystem = null;
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+  private AmazonS3 mS3Client = null;
+
+  @Before
+  public void before() throws Exception {
+    mLocalAlluxioClusterResource.start();
+    mFileSystem = mLocalAlluxioClusterResource.get().getClient();
+
+    mS3Client = AmazonS3ClientBuilder
+        .standard()
+        .withPathStyleAccessEnabled(true)
+        .withCredentials(
+            new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(mS3Proxy.getAccessKey(), mS3Proxy.getSecretKey())))
+        .withEndpointConfiguration(
+            new AwsClientBuilder.EndpointConfiguration(mS3Proxy.getUri().toString(),
+                Regions.US_WEST_2.getName()))
+        .build();
+    mS3Client.createBucket(TEST_BUCKET);
+  }
+
+  /**
+   * Writes a file through alluxio into UFS. Deletes the file from UFS.
+   * Read the file with sync interval setting to -1 should give the cached file.
+   * Read the file with sync interval setting to 0 should return error.
+   */
+  @Test
+  public void writeThenDeleteFromUfs() throws IOException, AlluxioException {
+    FileOutStream fos = mFileSystem.createFile(TEST_FILE_URI,
+        CreateFilePOptions.newBuilder().setOverwrite(true).build());
+    fos.write(TEST_CONTENT.getBytes());
+    fos.close();
+
+    mS3Client.deleteObject(TEST_BUCKET, TEST_FILE);
+    assertNotNull(mFileSystem.getStatus(TEST_FILE_URI));
+    FileInStream fis = mFileSystem.openFile(TEST_FILE_URI);
+    String content = new String(fis.readAllBytes());
```

Comment:
wrap the call inside a try-with-resource construct. There's some logic within the `FileInStream.close()` method and we want to cover that as well.